pandas/core/indexes/base.py

```diff
@@ -5563,7 +5563,9 @@ def _validate_join_method(method):
 def default_index(n):
     from pandas.core.indexes.range import RangeIndex
 
-    return RangeIndex(0, n, name=None)
+    if n == 0:
```

Comment:
so we are going to need to warn about the deprecation, you *might* be able to do this by passing a flag here, e.g. the higher level passes a flag if we need to warn (if n == 0)